src/Compiler.ts

```diff
@@ -42,9 +47,23 @@ function compile(fileNames: string[], options: ts.CompilerOptions): void {
         options.outDir = options.rootDir;
     }
 
+    if (!('addHeader' in options)) {
+        options.addHeader = true;
+    }
+
+    if (!options.luaLibDir) {
+        options.luaLibDir = options.outDir;
+    }
+    else {
+        options.luaLibDir = path.join(options.outDir, options.luaLibDir);
+    }
+    if (!fs.existsSync(options.luaLibDir)){
+        fs.mkdirSync(options.luaLibDir);
+    }
+
     // Copy lualib to target dir
     // This isnt run in sync because copyFileSync wont report errors.
-    fs.copyFile(path.resolve(__dirname, "../dist/lualib/typescript.lua"), path.join(options.outDir, "typescript_lualib.lua"), (err: NodeJS.ErrnoException) => {
+    fs.copyFile(path.resolve(__dirname, "../dist/lualib/typescript.lua"), path.join(options.luaLibDir, "typescript_lualib.lua"), (err: NodeJS.ErrnoException) => {
```

Comment:
When you change options.luaLibDir the require call should be resolved, in order to point to that new path, https://github.com/Perryvw/TypescriptToLua/blob/master/src/Transpiler.ts#L36

Reply:
See comment below.